### ISSUE TITLE
Pushing amplify appsync api changes

### ISSUE THREAD

**Issue by myfairshare** [aws-amplify/amplify-cli#4448]
**Which Category is your question related to?**
Pushing api changes

**Amplify CLI Version** 
4.21.0

**What AWS Services are you utilizing?**
amplify

**Provide additional details e.g. code snippets**
Hello,
   I am fairly new to AWS.  I have been building a project using amplify appsync and datastore.  Everything going ok but I was just working with just one model in my graphql schema to begin with.
I was following a Nader Dabit video to get off the ground.

I have since built up my schema a lot with more models and indexes so I have gone from one model to around 6 with a few @keys directives.

Today I tried to push my changes and got errors, then I realized in a recent Nader video he mentions pushing more than one @key addition causes errors.

So do I now need to strip back all my changes, push again, then push each new added model and one @key at a time, sounds quite time consuming, but I can live with that.

I have rolled back to my one original model at the moment but my local Indexdb now seems to be a bit out of whack after running my react front end again.  It seems to be still picking up the schema changes which I rolled back.

Hope this makes sense, thank you.

**Comment by nikhname** [comment]
@myfairshare after rolling back your model did you run ```amplify api gql-compile``` or do an ```amplify push```?

**Comment by myfairshare** [comment]
Hi Nik,

   Thanks for you reply.  So I commented out all my new models and @key directives in my gql schema and ran amplify push which gave no errors.  So I then added subsequent models one at a time minus directives and push them one by one, again no errors.  So I have just started adding @key directives one at a time and pushing, again no errors.  So I guess i will keep doing that.  I also deleted my few bits of data in the aws console, must go back and do for the datastore db also.  And of course my react code will need to be updated.  Is this bad practice, atleast I now know what to look out for.  I just hope my local indexdb works as before, thank you for any tips!
